File: lib/models/asset.dart
class Asset {
  final String symbol;
  final String name;
  final double price;
  final double change;
  final double volume;
  final double marketCap;
  final String market;
  final int iconColor;
  bool isInWatchlist;

  Asset({
    required this.symbol,
    required this.name,
    required this.price,
    required this.change,
    required this.volume,
    required this.marketCap,
    required this.market,
    required this.iconColor,
    this.isInWatchlist = false,
  });

  bool get isGain => change >= 0;

  Asset copyWith({
    double? price,
    double? change,
    double? volume,
    bool? isInWatchlist,
  }) {
    return Asset(
      symbol: symbol,
      name: name,
      price: price ?? this.price,
      change: change ?? this.change,
      volume: volume ?? this.volume,
      marketCap: marketCap,
      market: market,
      iconColor: iconColor,
      isInWatchlist: isInWatchlist ?? this.isInWatchlist,
    );
  }
}

File: Models/Asset.swift
import SwiftUI

struct Asset: Identifiable, Equatable, Hashable {
    var id: String { symbol }

    let symbol:    String
    let name:      String
    var price:     Double
    var change:    Double
    let volume:    Double
    let marketCap: Double
    let market:    String
    let iconHex:   String

    var isGain:    Bool  { change >= 0 }
    var iconColor: Color { Color(hex: iconHex) }

    var tradingViewSymbol: String {
        switch market {
        case "Crypto":
            return "BINANCE:\(symbol)USDT"
        case "Stocks":
            return "NASDAQ:\(symbol)"
        case "Forex":
            let clean = symbol.replacingOccurrences(of: "/", with: "")
            return "FX:\(clean)"
        default:
            return symbol
        }
    }

    static func == (lhs: Asset, rhs: Asset) -> Bool { lhs.symbol == rhs.symbol }
    func hash(into hasher: inout Hasher) { hasher.combine(symbol) }
}
